jetty-webapp/src/main/java/org/eclipse/jetty/webapp/WebInfConfiguration.java

```diff
@@ -716,6 +716,7 @@ public static String getCanonicalNameForWebAppTmpDir (WebAppContext context)
         Resource web_inf_lib = web_inf.addPath("/lib");
         if (web_inf_lib.exists() && web_inf_lib.isDirectory())
         {
+            // TODO should files be sorted? What is the interaction with Ordering?
```

Comment:
I believe it should be sorted here, mainly from the point of view of consistency.  If webappcontext sometimes applies an ordering to discovered jars, then it always should.

Reply:
Actually, cancel that - this is not in webappcontext...

@janbartel  can you have a look at this PR.   It looks like we might be scanning WEB-INF/lib twice - once for the classpath and once to find the resources list.     It would be great if this could be done only once.... but we have to also allow for other types of classloader.

Reply:
Yes, we do list the contents of WEB-INF/lib twice: once to find jars that we can apply the servlet ordering over, and once to find the jars that we populated onto the webapp classloader.

Discussed this with @gregw offine: for  jetty-9.4.x we should do the minimum and sort whenever we list the files in WEB-INF/lib. That is, in WebInfConfiguration.findWebInfLibJars() at line 717 we should sort the list of files returned, and also in WebAppClassLoader.addJars() at line 330. In jetty-10 we will remove the duplication of listing WEB-INF/lib files. For that, I've opened another bug: #1828.

@jglick would you like to modify the pull request as I've suggested for jetty-9.4.x?

Reply:
Seems prudent.